src/sources/kubernetes_logs/parser/cri.rs

```diff
@@ -27,86 +29,144 @@ pub const NEW_LINE_TAG: &str = "new_line_tag";
 #[derivative(Debug)]
 pub(super) struct Cri {
     #[derivative(Debug = "ignore")]
-    regex_parser: Box<dyn FunctionTransform>,
+    pattern: Regex,
+    capture_locations: CaptureLocations,
+    capture_names: Vec<(usize, String)>,
+    field: &'static str,
 }
 
-impl Cri {
-    /// Create a new [`Cri`] parser.
-    pub fn new(timezone: TimeZone) -> Self {
-        let regex_parser = {
-            let mut rp_config = RegexParserConfig::default();
+impl Default for Cri {
+    fn default() -> Self {
+        let pattern =
+            Regex::new(CRI_REGEX_PATTERN).expect("CRI log regex pattern should never fail");
```

Comment:
Would it make sense to move this initialization into a `OnceCell` and just use a reference?